### ISSUE TITLE
Untangle offline mode

### ISSUE THREAD

**Issue by SuperSandro2000** [go-gitea/gitea#31112]
### Description

Right now the situation around the offline mode and federated avatars is a bit of a mess especially after #28548. I've tried to understand the code around it and not noticed, that the offline mode isn't changing anything regarding to CDN but is just used as an override to Disable Gravatar and Enable Federated Avatars. I think it would be a lot less confusing if the offline mode setting would be just removed and the avatar features are controlled directly. Also since GRAVATAR_SOURCE exists, DISABLE_GRAVATAR can also be removed.

### Gitea Version

1.22

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

With NixOS ofcourse

### Database

PostgreSQL

**Comment by wxiaoguang** [comment]
Hmm yes, some options on the installation page and in the config file are totally a mess .... I totally agree with you 🤣 

They need to be refactored with a new design.

**Comment by Sharaf5** [comment]
that option was mainly to disable the cdn and just serve locally
I think it's still functional as I really notice no cdn actually
![Screenshot_20240527_223704](https://github.com/go-gitea/gitea/assets/36330265/1b2a371d-ba92-4698-9d3c-51e4c597e0ac)

**Comment by SuperSandro2000** [comment]
> that option was mainly to disable the cdn and just serve locally

I don't think that functionality exists anymore. I have offline mode turned off and all assets except for LibreAvatar and some extensions are served from Gitea itself.

**Comment by wxiaoguang** [comment]
That option is just wrong and is a no-op.

Will be removed by #36771
